### PR TITLE
Create table url type 77

### LINKED ISSUE #77: Create Table: url_type
### Overview

We need to create the url_type table so that we can update a shared data store across hackforla.org, vrms, civictechjobs, and tables (onboarding) project.

#### Details

A table and a model are the same thing

### Action Items

- [x] identify and document table description (see spreadsheet under Resources)
  - [x] if not, reach out to PD leads
- [x] compare and check off the data fields (below) against the ERD. Note any fields in the ERD that's missing from the list. (see Resources)
- [x] compare the check off associated tables (below) against the ERD. Note any associated tables in the ERD that's missing from the list (see Resources)
- [x] create a single model in Django (defining schema)
- [x] add seed data
- [x] write a test for the relationships this model will have with other models (e.g., creating a user and assigning them a set of permissions on a project).
- [x] write an API end point
- [x] write API unit tests
- [x] document the endpoint
- [ ] Release #41 from ice box

### Resources/Instructions

- See [People Depot Resources wiki page](https://github.com/hackforla/peopledepot/wiki/Resources-and-Links) for links
  - ERD
  - Table and Field Definitions
  - API Endpoint
- See the [wiki page](https://github.com/hackforla/peopledepot/wiki/Create-Table-issues-data-gathering-workflow) for instructions.

### Description

Type of the URL (ReadMe, Wiki, etc.)

### Data Fields

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] (PK) id - int - Record Id
   - [x] name - varchar
   - [x] description - textfield

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### Associated Tables

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] project_url (one-to-many)

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### PR DESCRIPTION
**Pull request by dmartin4820**
Fixes #77 

### What changes did you make?

- created url_type model
- register url_type in admin
- add url_type endpoints
- add url_type seed data
- update Create initial data scripts[¶](https://hackforla.github.io/peopledepot/how-to/create-initial-data-migrations/#create-initial-data-scripts) to clarify some of the seed data steps
- 
### Why did you make the changes (we will use this info to test)?

- To add a bit more clarity to adding seed data.

### CODE REVIEW

File: app/core/initial_data/UrlType_export.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        "name": "Wiki",
+        "description": "The wiki for the project"
+    },
+    {
+        "id": "7",
+        "name": "Test Site",
+        "description": "Development site used for testing"
+    },
+    {
+        "id": "88",
```
Comment (del9ra) [inline app/core/initial_data/UrlType_export.json:38]: I guess the id number was "8"?

**Review by del9ra — APPROVED** [review]
Code looks clean and works fine. Nice job!

File: app/core/initial_data/UrlType_export.json (hunk 1 above)
Comment (dmartin4820) [reply app/core/initial_data/UrlType_export.json:38]: Yeah, good catch. So this is a consequence of the id being set to 88 in the Excel sheet (as of writing this). You can see that [here](https://docs.google.com/spreadsheets/d/1x_zZ8JLS2hO-zG0jUocOJmX16jh-DF5dccrd_OEGNZ0/edit?gid=329100364#gid=329100364). I can make the update and re-push. Thanks

**Review by fyliu — CHANGES_REQUESTED** [review]
Thanks for working on this! I'm just commenting based on reading through the code changes for now.

Reviewing this brings up some existing problems I haven't made issues to deal with, so I'll do that. Please disregard the parts where I mention other issues. They're there to remind me what else needs to be documented. I try to put the relevant comments at the beginning.

File: app/core/admin.py [hunk 2]
```diff
@@ -245,3 +246,8 @@ class CheckTypeAdmin(admin.ModelAdmin):
 @admin.register(SocMajor)
 class SocMajorAdmin(admin.ModelAdmin):
     list_display = ("occ_code", "title")
+
+
+@admin.register(UrlType)
+class UrlType(admin.ModelAdmin):
```
[the remark this answers is not in the document]
Comment (dmartin4820) [reply app/core/admin.py:252]: Will fix. The docs are correct.

File: docs/how-to/create-initial-data-migrations.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ These are the steps:
 
 You must have Docker installed
 
-The initial data exists in a Google spreadsheet, such as [this one for People Depot][pd-data-spreadsheet]. There should be individual sheets named after the model names the data correspond to, such as `ProgramArea - Data`. The sheet name is useful for us to identify the model it corresponds to.
+The initial data exists in a Google spreadsheet, such as \[this one for People Depot\]\[pd-data-spreadsheet\]. There should be individual sheets named after the model names the data correspond to, such as `ProgramArea - Data`. The sheet name is useful for us to identify the model it corresponds to.
```
[the remark this answers is not in the document]
Comment (dmartin4820) [reply docs/how-to/create-initial-data-migrations.md]: Removed the changes made by mdformat. It didn't seem necessary to add the file to the pre-commit hook when I committed, and I'm not clear on how to add it to the pre-commit hook to check. I'm thinking that I need to add the file [here](https://github.com/hackforla/peopledepot/blob/main/.pre-commit-config.yaml#L141). Is that correct? If so, I can add that.

File: app/core/tests/test_models.py [hunk 4]
```diff
@@ -145,3 +145,9 @@ def test_check_type(check_type):
 
 def test_soc_major(soc_major):
     assert str(soc_major) == "Test Soc Major"
+
+
+def test_url_type(url_type):
+    assert str(url_type) == "This is a test url type name"
+    assert url_type.name == "This is a test url type name"
+    assert url_type.description == "This is a test url type description"
```
[the remark this answers is not in the document]
Comment (dmartin4820) [reply app/core/tests/test_models.py]: Removed both the asserts

File: app/core/admin.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmartin4820) [reply app/core/admin.py]: Fixed this. See [here](https://github.com/hackforla/peopledepot/pull/400/files#diff-b5f84ecfa193b6d3f4f450d0b8d1dd2fd09ba6b46758b1808eeb79f2d9fe8e41R252)

**Review by del9ra — APPROVED** [review]
Great job! Thank you for making the fixes.

**Review by fyliu — APPROVED** [review]
Great job! Thank you @dmartin4820!

**Comment by shmonks** [comment]
@fyliu it looks like you approved this - can you merge it or does something else need to be done?

**Pull request merged**